src/main/java/com/amazon/opendistroforelasticsearch/security/auth/UserInjector.java

```diff
@@ -66,13 +67,23 @@
 
     }
 
-    public static class InjectedUser extends User {
+    static class InjectedUser extends User {
         private transient TransportAddress transportAddress;
 
         public InjectedUser(String name) {
             super(name);
         }
 
+        private Object writeReplace() throws ObjectStreamException {
```

Comment:
What is this function used for ?

Reply:
It is used by Java serializer to substitute object for a different object.